api/handler_readme.go

```diff
@@ -0,0 +1,77 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/skeswa/gophr/common/errors"
+)
+
+const (
+	queryStringRepoTextKey = "r"
+	queryStringRefTextKey  = "hb"
+)
+
+var (
+	repoName string
+	ref      string
+)
+
+// ReadmeHandler creates an HTTP request handler that responds to fuzzy package
+// searches.
+func ReadmeHandler() func(http.ResponseWriter, *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+
+		// Get URL Query Params
+		qs := r.URL.Query()
+		if qs != nil {
+			// Get repo name
+			repoName = qs.Get(queryStringRepoTextKey)
```

Comment:
Wat. The example from the search handler has all transient state _inside_ the closure.